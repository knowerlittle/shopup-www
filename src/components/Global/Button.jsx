import React from 'react';
import styles from 'components/Global/button.css';
import PropTypes from 'prop-types';

export const options = {
  white: 'white',
  yellow: 'yellow',
  purple: 'purple',
  whiteYellow: 'white-yellow',
  disabled: 'disabled',
};

const getStyle = (color) => {
  const styleColor = options[color];
  return styles[styleColor];
};

export const setDimensions = (height, width) => ({
  height: `${height}px`,
  width: `${width}px`,
});

const Button = ({
  text,
  color,
  height,
  width,
  onClick,
}) => (
  <div>
    <div
      className={getStyle(color)}
      style={setDimensions(height, width)}
      onKeyDown={() => {}}
      role="presentation"
      onClick={onClick}
    >
      <p>{text}</p>
    </div>
  </div>
);

Button.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  height: PropTypes.string,
  width: PropTypes.string,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  height: '50',
  width: '200',
  onClick: () => ({}),
};

export default Button;
