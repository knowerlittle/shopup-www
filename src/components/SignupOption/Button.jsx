import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from 'components/SignupOption/index.css';

const Button = ({ text, link }) => (
  <div className={styles.buttonWrapper}>
    <Link to={link}>
      <div className={styles.rectangle}>
        <p className={styles.text}>{ text }</p>
      </div>
    </Link>
  </div>
);

Button.propTypes = {
  text: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
};

export default Button;
