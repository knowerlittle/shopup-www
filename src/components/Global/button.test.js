import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button, { options, setDimensions } from './Button';
import PropTypes from 'prop-types';

describe('<Button />', () => {
  describe('general', () => {
    it('options: has 5 color options', () => {
      const optionsCount = Object.keys(options).length;

      expect(optionsCount).toEqual(5);
      expect(options['white']).toBeTruthy();
      expect(options['yellow']).toBeTruthy();
      expect(options['purple']).toBeTruthy();
      expect(options['whiteYellow']).toBeTruthy();
      expect(options['disabled']).toBeTruthy();
    });

    it('setDimensions: returns correct object', () => {
      const result = setDimensions('100', '200');
      const expectedResult = {
        height: '100px',
        width: '200px',
      };
      expect(result).toEqual(expectedResult);
    });
  });

  describe('props', () => {
    const { text, color, height, width, onClick } = Button.propTypes;

    it('has 5', () => {
      const count = Object.keys(Button.propTypes).length;
      expect(count).toEqual(5);
    });
  
    it('\'text\' is a string that is required', () => {
      expect(text).toBe(PropTypes.string.isRequired);
    });
  
    it('\'color\' is a string that is required', () => {
      expect(color).toBe(PropTypes.string.isRequired);
    });
  
    it('\'height\' is a string', () => {
      expect(height).toBe(PropTypes.string);
    });
  
    it('\'width\' is a string', () => {
      expect(width).toBe(PropTypes.string);
    });
  
    it('\'onClick\' is a function', () => {
      expect(onClick).toBe(PropTypes.func);
    });
  });

  describe('defaultProps', () => {
    const { height, width, onClick } = Button.defaultProps;

    it('has 3', () => {
      const count = Object.keys(Button.defaultProps).length;
      expect(count).toEqual(3);
    });

    it('height is set to 50', () => {
      const heightDefault = '50';
      expect(height).toEqual(heightDefault);
    });

    it('width is set to 200', () => {
      const widthDefault = '200';
      expect(width).toEqual(widthDefault);
    });

    it('onClick returns empty object', () => {
      const onClickDefault = {};
      expect(onClick()).toEqual(onClickDefault);
    });
  });

  describe('Snapshot', () => {
    it('renders correctly', () => {
      const whiteTree = shallow(
        <Button 
          text='white button' 
          color='white'
          height='100'
          width='40'
        />
      );
      const yellowTree = shallow(
        <Button 
          text='yellow button' 
          color='yellow'
          height='300'
          width='30'
        />
      );

      expect(toJson(whiteTree)).toMatchSnapshot();
      expect(toJson(yellowTree)).toMatchSnapshot();
    });
  });
});