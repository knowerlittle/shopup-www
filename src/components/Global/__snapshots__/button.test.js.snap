// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshot renders correctly 1`] = `
<div>
  <div
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    style={
      Object {
        "height": "100px",
        "width": "40px",
      }
    }
  >
    <p>
      white button
    </p>
  </div>
</div>
`;

exports[`<Button /> Snapshot renders correctly 2`] = `
<div>
  <div
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    style={
      Object {
        "height": "300px",
        "width": "30px",
      }
    }
  >
    <p>
      yellow button
    </p>
  </div>
</div>
`;
